examples/gltf/src/main.rs

```diff
@@ -0,0 +1,156 @@
+use std::sync::Arc;
+
+fn load_gltf(
+    renderer: &rend3::Renderer,
+    path: &'static str,
+) -> (rend3::datatypes::MeshHandle, rend3::datatypes::MaterialHandle) {
+    let (doc, datas, _) = gltf::import(path).unwrap();
+    let mesh_data = doc.meshes().next().expect("no meshes in data.glb");
+
+    let primitive = mesh_data.primitives().next().expect("no primitives in data.glb");
+    let reader = primitive.reader(|b| Some(&datas.get(0)?.0[b.index()..b.length()]));
+    let mut normals = reader.read_normals();
+    let mut uv = reader.read_tex_coords(0).map(|tc| tc.into_f32());
+    let mut mesh = rend3::datatypes::Mesh {
+        vertices: reader
+            .read_positions()
+            .unwrap()
+            .map(|pos| rend3::datatypes::ModelVertex {
+                position: pos.into(),
+                normal: normals.as_mut().and_then(|n| n.next()).unwrap_or_default().into(),
+                uv: uv.as_mut().and_then(|n| n.next()).unwrap_or_default().into(),
+                color: [0; 4],
+            })
+            .collect(),
+        indices: reader.read_indices().unwrap().into_u32().collect(),
+    };
+    if normals.is_none() {
+        mesh.calculate_normals();
+    }
+
+    // Add mesh to renderer's world
+    let mesh_handle = renderer.add_mesh(mesh);
+
+    // Add basic material with all defaults except a single color.
+    let material = rend3::datatypes::Material {
+        albedo: rend3::datatypes::AlbedoComponent::Value(
+            primitive.material().pbr_metallic_roughness().base_color_factor().into(),
```

Comment:
We should factor the call to the material and the call to the pbr data into separate variables so it's easier to see how one might expand the example